src/gt4py/next/program_processors/runners/dace_fieldview/gtir_builtin_translators.py

```diff
@@ -219,11 +219,49 @@ def translate_as_fieldop(
     fun_node = node.fun
     assert len(fun_node.args) == 2
     stencil_expr, domain_expr = fun_node.args
-    assert isinstance(stencil_expr, gtir.Lambda)
-    assert isinstance(domain_expr, gtir.FunCall)
 
     # parse the domain of the field operator
     domain = extract_domain(domain_expr)
+    domain_index = sbs.Indices([dace_gtir_utils.get_map_variable(dim) for dim, _, _ in domain])
```

Comment:
There are (potentially) multiple indices here, so I would name the variable `domain_indices`.
This would also allow you to write loops such as
```python
for domain_index in domain_indices:
        ...
```